test/e2e/alertmanager_test.go

```diff
@@ -893,6 +893,90 @@ func assertLabelSelectorRequirement(reqs []metav1.LabelSelectorRequirement, must
 	return fmt.Errorf("required label selector %v not found in %v", mustInclude, reqs)
 }
 
+// TestAlertmanagerPlatformSecrets ensures secrets
+// are mounted correctly in Platform AlertManager container
+func TestAlertmanagerPlatformSecrets(t *testing.T) {
+	amSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-secret",
+			Namespace: f.Ns,
+			Labels: map[string]string{
+				"group":                    "amsecret-e2e",
+				framework.E2eTestLabelName: framework.E2eTestLabelValue,
+			},
+		},
+		Data: map[string][]byte{
+			"username": []byte("admin"),
+			"password": []byte("password"),
+		},
+	}
+
+	if err := f.OperatorClient.CreateIfNotExistSecret(ctx, amSecret); err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range []struct {
+		name        string
+		config      string
+		amName      string
+		amNamespace string
+	}{
+		{
+			name: "platform-alertmanager-secrets",
+			config: `alertmanagerMain:
+  secrets:
+  - test-secret`,
+			amName:      "main",
+			amNamespace: f.Ns,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			cm := &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterMonitorConfigMapName,
+					Namespace: f.Ns,
+					Labels: map[string]string{
+						framework.E2eTestLabelName: framework.E2eTestLabelValue,
+					},
+				},
+				Data: map[string]string{
+					"config.yaml": tc.config,
+				},
+			}
+			f.MustCreateOrUpdateConfigMap(t, cm)
+			t.Cleanup(func() {
+				f.MustDeleteConfigMap(t, cm)
+			})
+
```

Comment:
I would need to check whether reconciliation done before getting the alertmanager object here.